core/src/tasks/settle_match_internal.rs

```diff
@@ -0,0 +1,272 @@
+
+/// Describe the settle match internal task
+pub struct SettleMatchInternalTask {
+    /// The price at which the match was executed
+    pub execution_price: FixedPoint,
+    /// The identifier of the first order
+    pub order_id1: OrderIdentifier,
+    /// The identifier of the second order
+    pub order_id2: OrderIdentifier,
+    /// The validity proofs for the first order
+    pub order1_proof: OrderValidityProofBundle,
+    /// The validity proof witness for the first order
+    pub order1_validity_witness: OrderValidityWitnessBundle,
+    /// The validity proofs for the second order
+    pub order2_proof: OrderValidityProofBundle,
+    /// The validity proof witness for the second order
+    pub order2_validity_witness: OrderValidityWitnessBundle,
+    /// The match result
+    pub match_result: MatchResult,
+    /// The proof of `VALID MATCH MPC` generated in the first task step
+    pub valid_match_mpc: Option<ValidMatchMpcBundle>,
+    /// The starknet client to use for submitting transactions
+    pub starknet_client: StarknetClient,
+    /// A sender to the network manager's work queue
+    pub network_sender: TokioSender<GossipOutbound>,
+    /// A copy of the relayer-global state
+    pub global_state: RelayerState,
+    /// The work queue to add proof management jobs to
+    pub proof_manager_work_queue: CrossbeamSender<ProofManagerJob>,
+    /// The state of the task
+    pub task_state: SettleMatchInternalTaskState,
+}
+
+/// The state of the settle match internal task
+#[derive(Clone, Debug, Serialize)]
+pub enum SettleMatchInternalTaskState {
+    /// The task is awaiting scheduling
+    Pending,
+    /// The task is proving `VALID MATCH MPC` as a singleprover circuit
+    ProvingMatch,
+    /// The task is proving `VALID SETTLE`
+    ProvingSettle,
+    /// The task is submitting the match transaction
+    SubmittingMatch,
+    /// The task is updating the wallet state and Merkle openings
+    UpdatingState,
+    /// The task is updating validity proofs for the wallet
+    UpdatingValidityProofs,
+    /// The task has finished
+    Completed,
+}
+
+impl From<SettleMatchInternalTaskState> for StateWrapper {
+    fn from(value: SettleMatchInternalTaskState) -> Self {
+        Self::SettleMatchInternal(value)
+    }
+}
+
+impl Display for SettleMatchInternalTaskState {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        write!(f, "{self:?}")
+    }
+}
+
+/// The error type that the task emits
+#[derive(Clone, Debug, Serialize)]
+pub enum SettleMatchInternalTaskError {
+    /// Error enqueuing a job with another worker
+    EnqueuingJob(String),
+}
+
+impl Display for SettleMatchInternalTaskError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        write!(f, "{self:?}")
+    }
+}
+
+#[async_trait]
+impl Task for SettleMatchInternalTask {
+    type State = SettleMatchInternalTaskState;
+    type Error = SettleMatchInternalTaskError;
+
+    async fn step(&mut self) -> Result<(), Self::Error> {
```

Comment:
just a q: is this called repeatedly, i.e. poll-style? want to make sure we don't e.g. call `self.prove_match_mpc()` multiple times

Reply:
The `TaskDriver` calls this method in a loop, either to retry a failed step or to continue onto the next step. So the only way in which we'd call `prove_match_mpc()` multiple times is when it fails. See [`run_task_to_completion`](https://github.com/renegade-fi/renegade/blob/main/core/src/tasks/driver.rs#L141) for the driver code.